yt/frontends/art/io.py

```diff
@@ -121,7 +121,7 @@ def _get_field(self, field):
                 off = 1.0 / dd
                 tr[field] = rp(fields=[ax])[0] / dd - off
             if fname.startswith(f"particle_velocity_{ax}"):
-                (tr[field],) = rp(["v" + ax])
+                (tr[field],) = rp(fields=["v" + ax])
```

Comment:
I don't understand why this is a fix; I'm not denying it is, but I'd like to gain more insight. It's clearly an _improvement_ over using a partial with positional arguments, but given 4 out of 5 arguments of `read_particles` are filled in the partial def, I'd assume the fifth is the only one that can be passed to the function. In short I don't get why this causes a change in behaviour.

Reply:
I think there is some edge-case when mixing positional/non-positional arguments. With the old version, I get the following error:
```python
> import yt
… ds = yt.load_sample('D9p_500')
… ad = ds.all_data()
… ad['particle_velocity_x']"

Traceback (most recent call last):
  File "<string>", line 4, in <module>
  File "/home/ccc/Documents/prog/yt/yt/data_objects/data_containers.py", line 248, in __getitem__
    self.get_data(f)
  File "/home/ccc/Documents/prog/yt/yt/data_objects/selection_objects/data_selection_objects.py", line 203, in get_data
    particles, self, self._current_chunk
  File "/home/ccc/Documents/prog/yt/yt/geometry/geometry_handler.py", line 209, in _read_particle_fields
    chunks, selector, fields_to_read
  File "/home/ccc/Documents/prog/yt/yt/utilities/io_handler.py", line 204, in _read_particle_selection
    for field_r, vals in self._read_particle_fields(chunks, ptf, selector):
  File "/home/ccc/Documents/prog/yt/yt/frontends/art/io.py", line 101, in _read_particle_fields
    data = self._get_field((ptype, field))
  File "/home/ccc/Documents/prog/yt/yt/frontends/art/io.py", line 124, in _get_field
    (tr[field],) = rp(["v" + ax])
TypeError: read_particles() got multiple values for argument 'idxa'
```
To be honest, I am unsure why this is happening in the first place but I am sure the new behaviour is different!

Reply:
fair enough